src-self-hosted/astgen.zig

```diff
@@ -463,6 +470,47 @@ fn optionalType(mod: *Module, scope: *Scope, node: *ast.Node.SimplePrefixOp) Inn
     return addZIRUnOp(mod, scope, src, .optional_type, operand);
 }
 
+fn ptrType(mod: *Module, scope: *Scope, node: *ast.Node.PtrType) InnerError!*zir.Inst {
+    const tree = scope.tree();
+    const src = tree.token_locs[node.op_token].start;
+    const meta_type = try addZIRInstConst(mod, scope, src, .{
+        .ty = Type.initTag(.type),
+        .val = Value.initTag(.type_type),
+    });
+
+    const simple = node.ptr_info.allowzero_token == null and
+        node.ptr_info.align_info == null and
+        node.ptr_info.volatile_token == null and
+        node.ptr_info.sentinel == null;
+
+    if (simple) {
+        const child_type = try expr(mod, scope, .{ .ty = meta_type }, node.rhs);
```

Comment:
I was planning on putting 494 after all the other attributes so that they would get analyzed in the order that they appear in the source but I couldn't figure how to pass `child_type` to the sentinel cast result loc. If the order is not important then I'll just move it before the shortcut.

Reply:
Ahh that makes sense. Never mind my review comment then!

Reply:
That would still depend on being able to refer to the child type before the sentinel which doesn't seem possible but all the other attributes before the child type would still be an improvement.